File: java/src/test/java/io/binac/leetcode/FindMinimumInRotatedSortedArrayTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class FindMinimumInRotatedSortedArrayTest {

    private final FindMinimumInRotatedSortedArray.Solution1 solution1 = new FindMinimumInRotatedSortedArray.Solution1();

    private final int nums1[] = {3, 4, 5, 1, 2};

    private final int expected1 = 1;

    private final int nums2[] = {4, 5, 6, 7, 0, 1, 2};

    private final int expected2 = 0;

    @Test
    void test1() {
        assertEquals(expected1, solution1.findMin(nums1));
        assertEquals(expected2, solution1.findMin(nums2));
    }
}

File: cpp/tests/FindMinimumInRotatedSortedArrayTest.cpp
#include "FindMinimumInRotatedSortedArray.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(FindMinimumInRotatedSortedArray, Solution153_1) {
  std::vector<int> Nums1{3, 4, 5, 1, 2};
  const int Expected1 = 1;
  EXPECT_EQ(Expected1, Solution153_1().findMin(Nums1));

  std::vector<int> Nums2{4, 5, 6, 7, 0, 1, 2};
  const int Expected2 = 0;
  EXPECT_EQ(Expected2, Solution153_1().findMin(Nums2));
}
